### PR TITLE
added fetchAllFiles function to Folder.php

### PR DESCRIPTION
**Pull request by mnt**
Being able to get all the files in a folder is sometimes handy. This is done using a recursive search.

### CODE REVIEW

File: src/Krizalys/Onedrive/Folder.php [hunk 1]
```diff
@@ -65,4 +65,23 @@ public function createFolder($name, $description = null) {
...
 		return $this->_client->createFile($name, $this->_id, $content);
 	}
+
+	/**
+	 * Gets all files in the OneDrive folder referenced by this Folder instance.
+	 *
+ 	 * @return (array) The files in the OneDrive folder referenced by this
+	 *         Folder instance, as Object instances.
+	 */
+	public function fetchAllFiles(){
```
Comment (krizalys) [inline src/Krizalys/Onedrive/Folder.php]: Beware of the coding style: space before `{`.

File: src/Krizalys/Onedrive/Folder.php [hunk 2]
```diff
@@ -65,4 +65,23 @@ public function createFolder($name, $description = null) {
...
+		$files = [];
+
+		foreach($this->_client->fetchObjects($this->_id) as $file){
```
Comment (krizalys) [inline src/Krizalys/Onedrive/Folder.php]: Beware of the coding style: space after `foreach` and before `{`.

**Comment by krizalys** [comment]
Thanks for the PR.

I'm basically OK to add this functionality. However, for consistency reasons, I would prefer naming the function `fetchDescendantObjects`, since the 1-level function is called `fetchObjects`.

For the sake of completeness, you could also add a `fetchChildObjects`, which just call `fetchObjects` and add a deprecation notice to `fetchObjects` in favor of `fetchChildObjects`.

**Comment by mnt** [comment]
Okay I have made the changes. Feel free to tell me if there's anything else you would like changed :)

**Pull request merged**